operators/go/submariner_controller.go

```diff
@@ -0,0 +1,181 @@
+		IsController: true,
+		OwnerType:    &submarinerv1alpha1.Submariner{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileSubmariner implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileSubmariner{}
+
+// ReconcileSubmariner reconciles a Submariner object
+type ReconcileSubmariner struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a Submariner object and makes changes based on the state read
+// and what is in the Submariner.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileSubmariner) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Submariner")
+
+	// Fetch the Submariner instance
+	instance := &submarinerv1alpha1.Submariner{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Create submariner-engine SA
+	//subm_engine_sa := corev1.ServiceAccount{}
+	//subm_engine_sa.Name = "submariner-engine"
+	//reqLogger.Info("Created a new SA", "SA.Name", subm_engine_sa.Name)
+
+	// Define a new Pod object
+	// TODO: Make this responsive to size
+	pod := newPodForCR(instance)
+
+	// Set Submariner instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, pod, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Check if this Pod already exists
+	found := &corev1.Pod{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
+		err = r.client.Create(context.TODO(), pod)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// Pod created successfully - don't requeue
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Pod already exists - don't requeue
+	reqLogger.Info("Skip reconcile: Pod already exists", "Pod.Namespace", found.Namespace, "Pod.Name", found.Name)
+	return reconcile.Result{}, nil
```

Comment:
nit: reconcile.Result{} can be defined as a constant (I know this is from the scaffold. but its pretty silly repetition of it everywhere IMHO)

Reply:
I'm just going to stick with the scaffold, thanks.